### PR TITLE
feat: Adds script to sync standard-api version across all yaml files …

### PR DESCRIPTION
**Pull request by sgoossens**
…from package.json.

### CODE REVIEW

**Comment by sgoossens** [comment]
I still need to find the best way to get this to run in rush when updating.

File: repo-tools/scripts/synchronize-yaml.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ROOT=$(git rev-parse --show-toplevel)
+echo "Found repo root: ${ROOT}"
+
+# Get the version from package.json
+PACKAGE_VERSION=$(cat ${ROOT}/packages/*/package.json \
+  | grep @sixriver/standard-api \
+  | head -1 \
+  | awk -F: '{ print $2 }' \
+  | sed 's/[",]//g' \
+  | tr -d '[[:space:]]')
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:13]: `jq` can do this much more nicely for you

File: repo-tools/scripts/synchronize-yaml.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+YAML_FILES=$(git ls-files ${ROOT}/packages/oas/ \
+  | xargs -I {} find {} -type f -maxdepth 3 -name "*.yaml")
+
+# Now do the replacement in-place (MacOS/Unix compatible) https://frontend-apps.6river.org/api/v${PACKAGE_VERSION}
+for FILE in ${YAML_FILES}
+do
+  # If the yaml file imports from standard api, update the version
+  if grep -q "https://frontend-apps.6river.org/api/v" ${ROOT}/${FILE}
+  then
+    sed -i.bak -e "s#\(https://frontend-apps.6river.org/api/v\)[^/]*/#https://frontend-apps.6river.org/api/v${PACKAGE_VERSION}/#g" "${ROOT}/${FILE}"
```
Comment (sgoossens) [inline repo-tools/scripts/synchronize-yaml.sh:26]: is there a way around creating a .bak file and deleting it.  Without adding the file extension for the backup, the sed command fails

File: repo-tools/scripts/synchronize-yaml.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    rm ${ROOT}/${FILE}.bak
+  fi
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:28]: 1. style
2. quoting

```suggestion
  if grep -q "https://frontend-apps.6river.org/api/v" "${ROOT}/${FILE}" ; then
    sed -i.bak -e "s#\(https://frontend-apps.6river.org/api/v\)[^/]*/#https://frontend-apps.6river.org/api/v${PACKAGE_VERSION}/#g" "${ROOT}/${FILE}"
    rm "${ROOT}/${FILE}.bak"
  fi
```

File: repo-tools/scripts/synchronize-yaml.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  | grep @sixriver/standard-api \
+  | head -1 \
+  | awk -F: '{ print $2 }' \
+  | sed 's/[",]//g' \
+  | tr -d '[[:space:]]')
+echo "Extracted version: ${PACKAGE_VERSION}"
+
+# Find the yaml files in packages/oas tht are committed to git
+YAML_FILES=$(git ls-files ${ROOT}/packages/oas/ \
+  | xargs -I {} find {} -type f -maxdepth 3 -name "*.yaml")
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:18]: Quoting, Simpler:
```suggestion
YAML_FILES=$(git ls-files "${ROOT}/packages/oas/**/*.yaml")
```

File: repo-tools/scripts/synchronize-yaml.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+# Now do the replacement in-place (MacOS/Unix compatible) https://frontend-apps.6river.org/api/v${PACKAGE_VERSION}
+for FILE in ${YAML_FILES}
+do
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:22]: style:
```suggestion
for FILE in ${YAML_FILES} ; do
```

File: repo-tools/scripts/synchronize-yaml.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  # If the yaml file imports from standard api, update the version
+  if grep -q "https://frontend-apps.6river.org/api/v" ${ROOT}/${FILE}
+  then
+    sed -i.bak -e "s#\(https://frontend-apps.6river.org/api/v\)[^/]*/#https://frontend-apps.6river.org/api/v${PACKAGE_VERSION}/#g" "${ROOT}/${FILE}"
+    rm ${ROOT}/${FILE}.bak
+  fi
+done
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:29]: End files with a newline

File: repo-tools/scripts/synchronize-yaml.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (mgabeler-lee-6rs) [inline repo-tools/scripts/synchronize-yaml.sh:1]: make sure you get the "brew" bash on macOS so it isn't ancient bash:
```suggestion
#!/usr/bin/env bash
```

File: repo-tools/scripts/synchronize-yaml.sh (hunk 4 above)
Comment (mgabeler-lee-6rs) [reply repo-tools/scripts/synchronize-yaml.sh:18]: ugh ... I guess to do that, you need two expression for `git`:
`git ls-files "${ROOT}/packages/oas/*.yaml" "${ROOT}/packages/oas/**/*.yaml"`

File: repo-tools/scripts/synchronize-yaml.sh (hunk 1 above)
Comment (sgoossens) [reply repo-tools/scripts/synchronize-yaml.sh:14]: I have not looked into `jq` much other than basic usage.  I can update this as well

Comment (mgabeler-lee-6rs) [reply repo-tools/scripts/synchronize-yaml.sh:14]: I think something like this should work:
```shell
jq -r '.dependencies["@sixriver/standard-api"] | select(. != null)' packages/*/package.json | head -1
```

**Review by mgabeler-lee-6rs — APPROVED** [review]

**Pull request merged**